### ISSUE TITLE
test_shellcomp.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8586]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8586|
|Type|defect|
|Created|2016-07-11 09:40:46Z|

If I do this with Python 2:

```

python2 bin/trial twisted/python/test/test_shellcomp.py

twisted.python.test.test_shellcomp
  CompleterNotImplementedTests
    test_unknownShell ...                                                  [OK]
  EscapeTests
    test_escape ...                                                        [OK]
  ZshTests
    test_accumulateAdditionalOptions ...                                   [OK]
    test_accumulateMetadata ...                                            [OK]
    test_brokenActions ...                                                 [OK]
    test_incompleteCommandLine ...                                         [OK]
    test_incompleteCommandLine_case2 ...                                   [OK]
    test_incompleteCommandLine_case3 ...                                   [OK]
    test_mutuallyExclusiveCornerCase ...                                   [OK]
    test_optMethodsDontOverride ...                                        [OK]
    test_poorlyDescribedOptMethod ...                                      [OK]
    test_skipSubcommandList ...                                            [OK]
    test_verifyZshNames ...                                                [OK]
    test_zshCode ...                                                       [OK]
    test_zshCodeWithSubs ...                                               [OK]

-------------------------------------------------------------------------------
Ran 15 tests in 0.082s

PASSED (successes=15)
```

but if I try with Python 3:

```
python3 bin/trial twisted/python/test/test_shellcomp.py
twisted/python/test/test_shellcomp
  py ...                                                                [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted6/twisted/trial/runner.py", line 781, in loadByName
    return self.suiteFactory([self.findByName(name, recurse=recurse)])
  File "/Users/crodrigues/twisted6/twisted/trial/runner.py", line 682, in findByName
    return self.loadFile(_name, recurse=recurse)
  File "/Users/crodrigues/twisted6/twisted/trial/runner.py", line 652, in loadFile
    module = SourceFileLoader(name, fileName).load_module()
  File "<frozen importlib._bootstrap_external>", line 383, in _check_name_wrapper
    
  File "<frozen importlib._bootstrap_external>", line 805, in load_module
    
  File "<frozen importlib._bootstrap_external>", line 664, in load_module
    
  File "<frozen importlib._bootstrap>", line 268, in _load_module_shim
    
  File "<frozen importlib._bootstrap>", line 693, in _load
    
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
    
  File "<frozen importlib._bootstrap_external>", line 660, in exec_module
    
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "twisted/python/test/test_shellcomp.py", line 9, in <module>
    from cStringIO import StringIO
builtins.ImportError: No module named 'cStringIO'

twisted/python/test/test_shellcomp.py
-------------------------------------------------------------------------------
Ran 1 tests in 0.017s

FAILED (errors=1)

```

<details><summary>Searchable metadata</summary>

```
trac-id__8586 8586
type__defect defect
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468230046188152 1468230046188152
changetime__1468530907586552 1468530907586552
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_1"></a>|@acabhishek942 commented|
|-|-|

See : https://github.com/twisted/twisted/pull/325

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @acabhishek942|
|-|-|

Changes look good. Thanks.

Please add the new test to twisted/python/dist3.py file so that it will be executed by our automated tests.

Once that is done, please resubmit and I will merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)|@acabhishek942 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set status to `closed`|
|-|-|

In https://twistedmatrix.com/trac/changeset/a3447a91befecd92838a7d4950963cf6013f7d63

Merge pull request [#7522](https://github.com/twisted/twisted/issues/7522) from twisted/8586-acabhishek942-shellcomp
Author: acabhishek942
Reviewer: rodrigc
Fixes: [#8586](https://github.com/twisted/twisted/issues/8586)

Port _shellcomp.py to python 3
